### DISCUSSION TITLE
테스트 코드를 작성하는 이유를 설명해주세요.

### DISCUSSION THREAD

**Discussion by le2sky** [2024-woowacourse-study/level-interview#40] [Test]
개발자는 테스트 코드를 작성해서 무엇을 얻고자 하는가.

- 빠른 피드백
- 자동화
- 안정감

**Comment by kunsanglee** [comment]
- 버그 조기 발견: 개발 초기 단계에서 버그를 찾아 수정 비용과 시간을 절약합니다.
- 코드 품질 향상: 코드가 요구 사항을 충족하며 예상치 못한 문제 없이 동작하는지 확인합니다.
- 리팩토링 지원: 코드 구조 개선 시 기능 유지를 보장하며 안정성을 제공합니다.
- 개발 속도 향상: 코드의 정상 작동을 신속히 확인하여 개발자의 자신감을 높입니다.
- 문서화 역할: 기능 사용 방법과 기대 결과를 명확히 제시합니다.
- 고객 신뢰도 향상: 높은 품질의 제품을 제공하여 고객 만족을 보장합니다.

**Comment by seunghye218** [comment]
테스트를 작성하면 코드가 예상대로 작동하고 지정된 요구 사항을 충족하는지 확인하는 데 도움이 됩니다. 기존 기능이 손상될 염려 없이 코드를 리팩터링하고 개선할 수 있습니다. 

테스트를 통해 시간이 지남에 따라 코드베이스를 더 쉽게 이해하고 수정할 수 있으므로 기술적 부채와 향후 변경 비용이 줄어듭니다.

또한 자동화된 테스트는 CI/CD 의 중요한 구성 요소로 더 빠른 피드백 빈번하고 안정적인 배포를 가능하게 합니다.
